### PR TITLE
Integer overflow when parsing atom count in molecular formula

### PR DESCRIPTION
**Pull request by oatkachenko**

### CODE REVIEW

**Comment by johnmay** [comment]
It's the way it is for speed, I don't think the solution of using the slower API call and catching the exception is better, you still end up with a negative number. Nicer would be to just check if the value overflows

**Comment by oatkachenko** [comment]
I have replaced API call with manual overflow checking.

File: tool/formula/src/test/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulatorTest.java [hunk 1]
```diff
@@ -1483,4 +1483,19 @@ public void roundtrip(String mfStr, String expected) {
         roundtrip("[2H]2+2", "[[2]H2]2+");
         roundtrip("H+", "[H]+");
     }
+
+    @Test public void atomCountIsSafeForOverflow() {
+        // max integer
+        assertFormulaContainsExactlyAtoms(getMolecularFormula("H2147483647", builder), Integer.MAX_VALUE);
```
Comment (egonw) [inline tool/formula/src/test/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulatorTest.java:1489]: I don't understand this test. Should there not 2147483647 atoms here?

**Review by egonw — CHANGES_REQUESTED** [review]

File: tool/formula/src/test/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulatorTest.java (hunk 1 above)
Comment (johnmay) [reply tool/formula/src/test/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulatorTest.java:1489]: 2147483647 is Interger.MAX

**Comment by johnmay** [comment]
Looking back at this I think the simpler solution is to cap the atom count at some upper limit (e.g. 1 million) which is more than enough. TITIN is 35k amino acids https://en.wikipedia.org/wiki/Titin

**Comment by johnmay** [comment]
Now #808

**Pull request closed without merging**